barretenberg/cpp/src/barretenberg/dsl/acir_format/acir_to_constraint_buf.cpp

```diff
@@ -873,9 +881,31 @@ void handle_memory_op(Acir::Opcode::MemoryOp const& mem_op, BlockConstraint& blo
         block.type = BlockType::RAM;
     }
 
-    MemOp acir_mem_op = MemOp{ .access_type = access_type,
-                               .index = serialize_arithmetic_gate(mem_op.op.index),
-                               .value = serialize_arithmetic_gate(mem_op.op.value) };
+    // Update the ranges of the index using the array length
+    poly_triple index = serialize_arithmetic_gate(mem_op.op.index);
+    int bit_range = std::bit_width(block.init.size());
+    uint32_t index_witness = poly_to_witness(index);
+    if (index_witness != 0 && bit_range > 0) {
+        unsigned int u_bit_range = static_cast<unsigned int>(bit_range);
+        if (af.minimal_range.contains(index_witness)) {
+            if (af.minimal_range[index_witness] > u_bit_range) {
+                af.minimal_range[index_witness] = u_bit_range;
+            }
+        } else {
+            af.minimal_range[index_witness] = u_bit_range;
+        }
+        if (af.index_range.contains(index_witness)) {
+            if (af.index_range[index_witness] > u_bit_range) {
+                af.index_range[index_witness] = u_bit_range;
+            }
+        } else {
+            af.minimal_range[index_witness] = u_bit_range;
+        }
+        af.index_range[index_witness] = u_bit_range;
```

Comment:
We always want to set this? Wouldn't this override `af.index_range[index_witness]` if `af.index_range.contains(index_witness)` but `u_bit_range` is bigger than `af.index_range[index_witness]`. We would then replace a smaller range with a bigger one.

Reply:
Yes, thank you and sorry for the confusion.
I have also added a comment to explicit what the code is supposed to do.